Mathlib/GroupTheory/Complement.lean

```diff
@@ -333,6 +333,140 @@ theorem exists_right_transversal (g : G) : ∃ S ∈ rightTransversals (H : Set
 #align subgroup.exists_right_transversal Subgroup.exists_right_transversal
 #align add_subgroup.exists_right_transversal AddSubgroup.exists_right_transversal
 
+namespace IsComplement
+
+/-- The equivalence `G ≃ S × T`, such that the inverse is  `(*) : S × T → G` -/
+noncomputable def equiv {S T : Set G} (hST : IsComplement S T) : G ≃ S × T :=
+  (Equiv.ofBijective (fun x : S × T => x.1.1 * x.2.1) hST).symm
+
+variable (hST : IsComplement S T) (hHT : IsComplement H T) (hSK : IsComplement S K)
+
+theorem equiv_symm_apply (x : S × T) : (hST.equiv.symm x : G) = x.1.1 * x.2.1 := rfl
+
+@[simp]
+theorem equiv_fst_mul_equiv_snd (g : G) : ↑(hST.equiv g).fst * (hST.equiv g).snd = g :=
+  (Equiv.ofBijective (fun x : S × T => x.1.1 * x.2.1) hST).right_inv g
+
+theorem equiv_fst_eq_mul_inv (g : G) : ↑(hST.equiv g).fst = g * ((hST.equiv g).snd : G)⁻¹ :=
+  eq_mul_inv_of_mul_eq (hST.equiv_fst_mul_equiv_snd g)
+
+theorem equiv_snd_eq_inv_mul (g : G) : ↑(hST.equiv g).snd = ((hST.equiv g).fst : G)⁻¹ * g :=
+  eq_inv_mul_of_mul_eq (hST.equiv_fst_mul_equiv_snd g)
+
+theorem equiv_fst_eq_of_leftCosetEquivalence {g₁ g₂ : G}
+    (h : LeftCosetEquivalence K g₁ g₂) : (hSK.equiv g₁).fst = (hSK.equiv g₂).fst := by
+  apply Subtype.val_injective
+  rcases mem_leftTransversals_iff_existsUnique_inv_mul_mem.1 hSK g₁ with ⟨x, hx⟩
+  dsimp at hx
+  rw [hx.2 (hSK.equiv g₁).fst (hSK.equiv_snd_eq_inv_mul g₁ ▸ (hSK.equiv g₁).snd.prop),
+    hx.2 (hSK.equiv g₂).fst]
+  simp only [LeftCosetEquivalence, leftCoset, Set.image_mul_left, Set.ext_iff,
+    Set.mem_preimage, SetLike.mem_coe] at h
+  rw [SetLike.mem_coe, ← Subgroup.inv_mem_iff, mul_inv_rev, inv_inv, h,
+    ← Subgroup.inv_mem_iff, mul_inv_rev, inv_inv, ← hSK.equiv_snd_eq_inv_mul g₂]
+  exact Subtype.property _
+
+theorem equiv_snd_eq_of_rightCosetEquivalence {g₁ g₂ : G}
```

Comment:
```suggestion
theorem RightCosetEquivalence.equiv_snd {g₁ g₂ : G}
```
or something like that?

Reply:
It's already in the `IsComplement` namespace which is sensible, otherwise `equiv` could be anything.

Reply:
Ah I see. What about `RightCosetEquivalent.isComplementEquiv_snd`? Hmm... that's not much shorter.

Reply:
It's also nice because I can use the dot with my `IsComplement` hypothesis.